### PR TITLE
Tape value

### PR DESCRIPTION
**Pull request by Licenser**
Add dom style access to `Tape`

### CODE REVIEW

File: src/value/tape/cmp.rs [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+impl<'tape, 'input> PartialEq<f64> for Value<'tape, 'input> {
+    #[inline]
+    #[must_use]
+    fn eq(&self, other: &f64) -> bool {
+        self.as_f64().map(|t| t.eq(other)).unwrap_or_default()
+    }
+}
+
+// impl<'tape, 'input, K, T, S> PartialEq<std::collections::HashMap<K, T, S>> for Value<'tape, 'input>
```
Comment (darach) [inline src/value/tape/cmp.rs:167]: Remove or add a comment on the value of keeping this in commented out

File: src/value/tape.rs [hunk 2]
```diff
@@ -1,8 +1,29 @@
...
+impl<'input> Tape<'input> {
+    /// FIXME: add docs
+    #[must_use]
+    pub fn as_value(&self) -> Value<'_, 'input> {
+        // Skip initial zero
+        Value(&self.0[1..])
+    }
+}
+
+/// Wrapper around the tape that allows interacting with it via a `Value`-like API.
```
Comment (darach) [inline src/value/tape.rs:23]:
```suggestion
/// Wrapper around the tape that allows interaction via a `Value`-like API.
```

File: src/serde.rs [hunk 3]
```diff
@@ -62,10 +62,30 @@ where
     let mut deserializer = stry!(Deserializer::from_slice(s));
     T::deserialize(&mut deserializer)
 }
-/// parses a str  using a serde deserializer.
+
+/// parses a byte slice using a serde deserializer.
```
Comment (darach) [inline src/serde.rs:66]:
```suggestion
/// Parses a byte slice using a serde deserializer.
```

File: src/serde.rs [hunk 4]
```diff
@@ -62,10 +62,30 @@ where
...
+#[cfg_attr(not(feature = "no-inline"), inline)]
+pub fn from_slice_with_buffers<'a, T>(s: &'a mut [u8], buffers: &mut Buffers) -> Result<T>
+where
+    T: Deserialize<'a>,
+{
+    let mut deserializer = stry!(Deserializer::from_slice_with_buffers(s, buffers));
+    T::deserialize(&mut deserializer)
+}
+
+/// parses a str using a serde deserializer.
```
Comment (darach) [inline src/serde.rs:83]:
```suggestion
/// Parses a str using a serde deserializer.
```

File: src/serde.rs [hunk 5]
```diff
@@ -62,10 +62,30 @@ where
...
 /// note that the slice will be rewritten in the process and
 /// might not remain a valid utf8 string in its entirety.
 ///
+/// It is adviced to use `from_slice` instead.
```
Comment (darach) [inline src/serde.rs:87]:
```suggestion
/// It is recommended to use `from_slice` instead.
```

File: src/serde.rs [hunk 6]
```diff
@@ -85,6 +105,36 @@ where
     T::deserialize(&mut deserializer)
 }
 
+/// parses a str using a serde deserializer.
```
Comment (darach) [inline src/serde.rs:108]:
```suggestion
/// Parses a str using a serde deserializer.
```

File: src/serde.rs [hunk 7]
```diff
@@ -105,6 +155,28 @@ where
     T::deserialize(&mut deserializer)
 }
 
+/// parses a Reader using a serde deserializer.
```
Comment (darach) [inline src/serde.rs:158]:
```suggestion
/// Parses a Reader using a serde deserializer.
```

File: src/value/tape/cmp.rs (hunk 1 above)
Comment (Licenser) [reply src/value/tape/cmp.rs:167]: good call this can even be commented back in now

**Pull request merged**